File: 1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.java
class Solution {
    public int isPrefixOfWord(String sentence, String searchWord) {
        String[] words = sentence.split(" ");
        for (int idx = 0; idx < words.length; idx++) {
            if (words[idx].startsWith(searchWord)) {
                return idx + 1; // Return 1-based index
            }
        }
        return -1;
    }
}

File: 1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1455-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        int idx = 1;
        for(int i = 0 ; i < sentence.size();i++){
            int j = 0;
            while(j < searchWord.size() && searchWord[j] == sentence[i])i++,j++;
            if(j == searchWord.size())return idx;
            while(i < sentence.size() && sentence[i]!=' ')i++;
            idx++;
        }
        return -1;
    }
};